packages/react-flicking/src/react-flicking/CrossFlicking.tsx

```diff
@@ -0,0 +1,135 @@
+ */
+import * as React from "react";
+import ListDiffer from "@egjs/list-differ";
+import {
+  CrossFlicking as VanillaCrossFlicking,
+  NormalRenderingStrategy,
+  CrossFlickingEvent,
+  HoldStartEvent,
+  HoldEndEvent,
+  MoveStartEvent,
+  MoveEvent,
+  MoveEndEvent,
+  WillChangeEvent,
+  ChangedEvent,
+  EVENTS,
+  SIDE_EVENTS,
+  CrossFlickingOptions,
+  FlickingOptions,
+} from "@egjs/flicking";
+
+import { DEFAULT_PROPS } from "./consts";
+import { FlickingProps } from "./types";
+import ReactRenderer, { ReactRendererOptions } from "./ReactRenderer";
+import ReactElementProvider from "./ReactElementProvider";
+import Flicking from "./Flicking";
+
+export interface CrossFlickingProps extends FlickingProps {
+  onSideHoldStart: (e: CrossFlickingEvent<HoldStartEvent<Flicking>>) => any;
+  onSideHoldEnd: (e: CrossFlickingEvent<HoldEndEvent<Flicking>>) => any;
+  onSideMoveStart: (e: CrossFlickingEvent<MoveStartEvent<Flicking>>) => any;
+  onSideMove: (e: CrossFlickingEvent<MoveEvent<Flicking>>) => any;
+  onSideMoveEnd: (e: CrossFlickingEvent<MoveEndEvent<Flicking>>) => any;
+  onSideWillChange: (e: CrossFlickingEvent<WillChangeEvent<Flicking>>) => any;
+  onSideChanged: (e: CrossFlickingEvent<ChangedEvent<Flicking>>) => any;
+}
+
+export const CROSSFLICKING_DEFAULT_PROPS: CrossFlickingProps = {
+  ...DEFAULT_PROPS,
+  onSideHoldStart: (e: CrossFlickingEvent<HoldStartEvent>) => {},
+  onSideHoldEnd: (e: CrossFlickingEvent<HoldEndEvent>) => {},
+  onSideMoveStart: (e: CrossFlickingEvent<MoveStartEvent>) => {},
+  onSideMove: (e: CrossFlickingEvent<MoveEvent>) => {},
+  onSideMoveEnd: (e: CrossFlickingEvent<MoveEndEvent>) => {},
+  onSideWillChange: (e: CrossFlickingEvent<WillChangeEvent>) => {},
+  onSideChanged: (e: CrossFlickingEvent<ChangedEvent>) => {},
+};
+
+class CrossFlicking extends Flicking {
+  public static defaultProps: CrossFlickingProps = CROSSFLICKING_DEFAULT_PROPS;
+
+  public constructor(
+    props: Partial<
+      FlickingProps &
+        FlickingOptions &
+        CrossFlickingOptions &
+        CrossFlickingProps
+    >
+  ) {
+    super(props);
+  }
+
+  public componentDidMount() {
+    const props = this.props as Required<CrossFlickingProps>;
+    const rendererOptions: ReactRendererOptions = {
+      reactFlicking: this,
+      strategy: new NormalRenderingStrategy({
+        providerCtor: ReactElementProvider,
+      }),
+    };
+
+    const flicking = new VanillaCrossFlicking(this._viewportElement, {
+      ...props,
+      externalRenderer: new ReactRenderer(rendererOptions),
+    });
+
+    this._vanillaFlicking = flicking;
+
+    const children = this._getChildren();
+    this._jsxDiffer = new ListDiffer(children, (panel) => panel.key!);
+    this._pluginsDiffer = new ListDiffer<any>();
+    this._prevChildren = children;
+
+    this._bindEvents();
+    this._checkPlugins();
+
+    if (props.status) {
+      flicking.setStatus(props.status);
+    }
+  }
+
+  public render() {
+    const props = this.props;
```

Comment:
It seems like you'll need to filter only the parts you need from Cross's props.

Reply:
Added logic to filter props before passing them to the CrossFlicking instance.